core/src/smartcontracts/isi/account.rs

```diff
@@ -17,53 +17,61 @@ pub mod isi {
 
     impl<W: WorldTrait> Execute<W> for Mint<Account, PublicKey> {
         type Error = Error;
-        type Diff = DataEvent;
 
         #[metrics(+"mint_account_pubkey")]
         fn execute(
             self,
             _authority: <Account as Identifiable>::Id,
             wsv: &WorldStateView<W>,
-        ) -> Result<Self::Diff, Self::Error> {
-            let public_key = self.object.clone();
+        ) -> Result<Vec<DataEvent>, Self::Error> {
```

Comment:
As per the overall comment, I think that `DataEvents` should not be emitted by the execution, but instead be emitted by the interior mutation of the WSV. 

The benefits are:
1) No programmer error, if somebody modifies the WSV, but forgets to return the event inside Iroha 2. 
2) No malicious attack, when a WASM smartcontract deliberately doesn't return some events to avoid some triggers.

Reply:
I've been thinking about this, and I agree with this point. The only reason why I still went with returning `DataEvent`'s is because a dev adding a new instruction might forget to emit events from inside the `execute` method and I could think of no other pattern that would alleviate this concern. 

So the only reason to return data events was because I think it's more robust. Maybe silly?

Reply:
We shouldn't trust the ISI to emit data events at all. The WSV has interior mutability, so we should attach even emission to the mutators directly.

Reply:
now I get what you're saying. Yes, this makes total sense

Reply:
> 1. No programmer error, if somebody modifies the WSV, but forgets to return the event inside Iroha 2.

What if somebody modifies the WSV without going through `Executable`?
That means WSV is not reproducible from committed blocks and lacks the consensus.
We should not allow that possibility

> a dev adding a new instruction might forget to emit events from inside the `execute` method

`Diff` is bounded to impl `Into<Vec<DataEvent>>` though

Reply:
This discussion is a little beyond the scope of the PR. I think it's ready to merge, and that we should continue the discussion in the chat.